### PR TITLE
feat: debug lwc tests

### PR DESCRIPTION
**Pull request by CristiCanizales**
Implement tests for debugging LWC tests
[W-14866666](https://gus.lightning.force.com/a07EE00001ieHCuYAM)

Test run: https://github.com/forcedotcom/salesforcedx-vscode/actions/runs/9036615776/job/24833842612 ✅

### CODE REVIEW

File: test/utilities/sideBar.ts [hunk 1]
```diff
@@ -118,25 +118,26 @@ export async function getVisibleItems(
   return [...rows.values()];
 }
 
-export async function retrieveAllApexTestItemsFromSidebar(
+export async function retrieveAllTestItemsFromSidebar(
```
Comment (peternhale) [inline test/utilities/sideBar.ts:121]: Good refactor - thanks

File: test/utilities/sideBar.ts [hunk 2]
```diff
@@ -118,25 +118,26 @@ export async function getVisibleItems(
...
 
   // If the Apex tests did not show up, click the refresh button on the top right corner of the Test sidebar
   for (let x = 0; x < 3; x++) {
-    if (apexTestsItems.length === 1) {
-      await apexTestsSection.elem.click();
-      const refreshAction = await apexTestsSection.getAction('Refresh Tests');
+    if (testsItems.length === 1) {
+      await testsSection.elem.click();
+      const refreshAction = await testsSection.getAction(actionLabel);
       await refreshAction!.elem.click();
```
Comment (peternhale) [inline test/utilities/sideBar.ts:134]: Avoid `!`

File: test/utilities/sideBar.ts (hunk 1 above)
Comment (peternhale) [reply test/utilities/sideBar.ts:121]: One nit, function name `retrieveAllTestItemsFromSidebar` implies that all tests in the section are being fetched, while the implementation will stop the fetch when the number of tests found === expectedNumTests.

Is there a better name for the function?

**Review by peternhale — CHANGES_REQUESTED** [review]

File: test/specs/debugLwcTests.e2e.ts [hunk 3]
```diff
@@ -0,0 +1,291 @@
...
+    const lwcTestItem = (await lwcTestsSection.findItem('lwc1')) as TreeItem;
+    await lwcTestItem.select();
+    const debugTestsAction = await lwcTestItem.getActionButton(
+      'SFDX: Debug Lightning Web Component Test File'
+    );
+    await debugTestsAction!.elem.click();
+    await utilities.pause(10);
+
+    // Continue with the debug session
+    await browser.keys(['F5']);
```
[the remark this answers is not in the document]
Comment (CristiCanizales) [reply test/specs/debugLwcTests.e2e.ts:87]: This is like clicking the continue button to continue debugging

[the remark this answers is not in the document]
Comment (peternhale) [reply test/specs/debugLwcTests.e2e.ts:87]: Cool, would it make sense for those that follow to use a constant that names what is happening in the UI?

`await browser.keys([CONTINUE]);`

[the remark this answers is not in the document]
Comment (CristiCanizales) [reply test/specs/debugLwcTests.e2e.ts:87]: Sure! Good idea

**Review by peternhale — APPROVED** [review]

**Pull request merged**